### PR TITLE
Optimize maths: Use multiply instead of divide

### PR DESCRIPTION
**Pull request by magicrub**
Float multiply is faster than float divide and integer divide. Confirmed by @WickedShell 

This PR replaces most instances of:
`"/10" and "/100" and "/1000" `
 with
`"*0.1f" and *0.01f" and *0.001f".`

Exceptions:
- anything that looks like it might depend on integer roundoff like RTC (min,sec,hour)
- anything 64bit
- anything that was followed with a % operation
- anything larger than /1000
- anything that had an arithmetic OR/AND operation

### CODE REVIEW

**Comment by khancyr** [comment]
you could have also unify the syntax ;-P

File: APMrover2/AP_MotorsUGV.cpp [hunk 1]
```diff
@@ -843,7 +843,7 @@ float AP_MotorsUGV::get_scaled_throttle(float throttle) const
 
     // calculate scaler
     const float sign = (throttle < 0.0f) ? -1.0f : 1.0f;
-    const float throttle_pct = constrain_float(throttle, -100.0f, 100.0f) / 100.0f;
+    const float throttle_pct = constrain_float(throttle, -100.0f, 100.0f) * 0.01f;
```
Comment (peterbarker) [inline APMrover2/AP_MotorsUGV.cpp]: So unrelated to your patch here, but this variable has no right to call itself a percentage.

File: libraries/AP_GPS/AP_GPS_MAV.cpp [hunk 2]
```diff
@@ -114,7 +114,7 @@ void AP_GPS_MAV::handle_msg(const mavlink_message_t *msg)
             mavlink_msg_hil_gps_decode(msg, &packet);
 
             state.time_week = 0;
-            state.time_week_ms  = packet.time_usec/1000;
```
Comment (peterbarker) [inline libraries/AP_GPS/AP_GPS_MAV.cpp]: This doesn't look right.

Comment (magicrub) [reply libraries/AP_GPS/AP_GPS_MAV.cpp]: oh, is that 64bit? Then it should be removed.

**Comment by magicrub** [comment]
@peterbarker  can you point out some of the behavior changes?

**Comment by tridge** [comment]
several of the changes involve changing integer maths to float maths. I don't see how that is justified

File: AntennaTracker/tracking.cpp [hunk 3]
```diff
@@ -139,9 +139,9 @@ void Tracker::tracking_update_position(const mavlink_global_position_int_t &msg)
 {
     vehicle.location.lat = msg.lat;
     vehicle.location.lng = msg.lon;
-    vehicle.location.alt = msg.alt/10;
```
Comment (tridge) [inline AntennaTracker/tracking.cpp]: these two were integer maths

File: ArduCopter/GCS_Mavlink.cpp [hunk 4]
```diff
@@ -1166,12 +1166,12 @@ void GCS_MAVLINK_Copter::handleMessage(mavlink_message_t* msg)
         Location loc;
         loc.lat = packet.lat;
         loc.lng = packet.lon;
-        loc.alt = packet.alt/10;
```
Comment (tridge) [inline ArduCopter/GCS_Mavlink.cpp]: integer maths

File: ArduCopter/GCS_Mavlink.cpp [hunk 5]
```diff
@@ -1166,12 +1166,12 @@ void GCS_MAVLINK_Copter::handleMessage(mavlink_message_t* msg)
...
+        loc.alt = packet.alt*0.1f;
         Vector3f vel(packet.vx, packet.vy, packet.vz);
         vel *= 0.01f;
 
         gps.setHIL(0, AP_GPS::GPS_OK_FIX_3D,
-                   packet.time_usec/1000,
```
Comment (tridge) [inline ArduCopter/GCS_Mavlink.cpp]: integer maths

File: ArduPlane/GCS_Mavlink.cpp [hunk 6]
```diff
@@ -1121,15 +1121,15 @@ void GCS_MAVLINK_Plane::handleMessage(mavlink_message_t* msg)
         last_hil_state = packet;
 
         // set gps hil sensor
-        const Location loc{packet.lat, packet.lon, packet.alt/10, Location::AltFrame::ABSOLUTE};
```
Comment (tridge) [inline ArduPlane/GCS_Mavlink.cpp]: integer maths

File: ArduPlane/commands_logic.cpp [hunk 7]
```diff
@@ -741,7 +741,7 @@ bool Plane::verify_loiter_to_alt(const AP_Mission::Mission_Command &cmd)
         if (labs(loiter.sum_cd) > 1 && (loiter.reached_target_alt || loiter.unable_to_acheive_target_alt)) {
             // primary goal completed, initialize secondary heading goal
             if (loiter.unable_to_acheive_target_alt) {
-                gcs().send_text(MAV_SEVERITY_INFO,"Loiter to alt was stuck at %d", current_loc.alt/100);
```
Comment (tridge) [inline ArduPlane/commands_logic.cpp]: this introduces a bug

File: ArduPlane/takeoff.cpp [hunk 8]
```diff
@@ -210,7 +210,7 @@ int16_t Plane::get_takeoff_pitch_min_cd(void)
                 relative_alt_cm >= 1000 &&
                 sec_to_target <= g.takeoff_pitch_limit_reduction_sec) {
                 // make a note of that altitude to use it as a start height for scaling
-                gcs().send_text(MAV_SEVERITY_INFO, "Takeoff level-off starting at %dm", remaining_height_to_target_cm/100);
+                gcs().send_text(MAV_SEVERITY_INFO, "Takeoff level-off starting at %dm", remaining_height_to_target_cm*0.01f);
```
Comment (tridge) [inline ArduPlane/takeoff.cpp]: another bug introduced

File: ArduSub/commands_logic.cpp [hunk 9]
```diff
@@ -482,7 +482,7 @@ void Sub::do_nav_delay(const AP_Mission::Mission_Command& cmd)
         // absolute delay to utc time
         nav_delay_time_max = AP::rtc().get_time_utc(cmd.content.nav_delay.hour_utc, cmd.content.nav_delay.min_utc, cmd.content.nav_delay.sec_utc, 0);
     }
-    gcs().send_text(MAV_SEVERITY_INFO, "Delaying %u sec",(unsigned int)(nav_delay_time_max/1000));
```
Comment (tridge) [inline ArduSub/commands_logic.cpp]: and another bug

File: libraries/AP_AHRS/AP_AHRS_DCM.cpp [hunk 10]
```diff
@@ -198,7 +198,7 @@ AP_AHRS_DCM::reset(bool recover_eulers)
         yaw = radians(yaw_sensor*0.01f);
         _dcm_matrix.from_euler(roll, pitch, yaw);
         gcs().send_text(MAV_SEVERITY_INFO, "Restored watchdog attitude %d %d %d",
-                        roll_sensor/100, pitch_sensor/100, yaw_sensor/100);
+                        roll_sensor*0.01f, pitch_sensor*0.01f, yaw_sensor*0.01f);
```
Comment (tridge) [inline libraries/AP_AHRS/AP_AHRS_DCM.cpp]: more bugs

**Comment by magicrub** [comment]
How about we change only the ones that are already float?

**Pull request closed without merging**